src/widgets/voice-search/voice-search.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { render, unmountComponentAtNode } from 'preact-compat';
+import cx from 'classnames';
+import {
+  getContainerNode,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import { component } from '../../lib/suit';
+import connectVoiceSearch, {
+  VoiceSearchRenderer,
+} from '../../connectors/voice-search/connectVoiceSearch';
+import VoiceSearch from '../../components/VoiceSearch/VoiceSearch';
+import defaultTemplates from './defaultTemplates';
+import { Template } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({ name: 'voice-search' });
+const suit = component('VoiceSearch');
+
+export type VoiceSearchCSSClasses = {
+  root: string;
+  button: string;
+  status: string;
+};
+
+export type VoiceSearchTemplates = {
+  buttonText: Template<object>;
+  status: Template<object>;
+};
+
+export type VoiceListeningState = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  errorCode?: string;
+};
+
+type VoiceSearchWidgetParams = {
+  container: string | HTMLElement;
+  cssClasses: VoiceSearchCSSClasses;
+  templates: VoiceSearchTemplates;
+  searchAsYouSpeak?: boolean;
+};
+
+type VoiceSearchWidgetRenderParams = {
+  isSupportedBrowser: boolean;
+  isListening: boolean;
+  toggleListening: (toggleListeningParams: {
+    searchAsYouSpeak: boolean;
+  }) => void;
+  voiceListeningState: VoiceListeningState;
+  widgetParams: VoiceSearchWidgetParams;
+};
+
+const renderer: VoiceSearchRenderer<VoiceSearchWidgetParams> = ({
+  isSupportedBrowser,
+  isListening,
+  toggleListening,
+  voiceListeningState,
+  widgetParams,
+}: VoiceSearchWidgetRenderParams) => {
```

Comment:
You don't have to provide the type, values are already inferred from the `VoiceSearchRenderer`.

Reply:
Done